stateproof/db_test.go

```diff
@@ -117,3 +116,100 @@ func TestPendingSigDB(t *testing.T) {
 		}
 	}
 }
+
+func TestSigExistQuery(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	dbs, _ := dbOpenTest(t, true)
+	defer dbs.Close()
+
+	require.NoError(t, makeStateProofDB(dbs.Wdb))
+
+	n := 5
+	var accts []basics.Address
+	// setup:
+	for r := basics.Round(0); r < basics.Round(n); r++ {
+		var psig pendingSig
+		crypto.RandBytes(psig.signer[:])
+		accts = append(accts, psig.signer)
+
+		require.NoError(t, dbs.Wdb.Atomic(func(ctx context.Context, tx *sql.Tx) error {
+			return addPendingSig(tx, r, psig)
+		}))
+	}
+
+	// test:
+	for r := basics.Round(0); r < basics.Round(n); r++ {
+		require.NoError(t, dbs.Wdb.Atomic(func(ctx context.Context, tx *sql.Tx) error {
+			exists, err := isPendingSigExist(tx, r, accts[r])
+			require.NoError(t, err)
+			require.True(t, exists)
+			return nil
+		}))
+
+		require.NoError(t, dbs.Wdb.Atomic(func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
maybe we can remove the user's signature from the database and make sure  `isPendingSigExist` returns false (instead of invalid address)